### PR TITLE
images

### PR DESCRIPTION
**Pull request by Borislav466**

### CODE REVIEW

File: src/api/images.api.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import BaseApi from './base.api';
 
 export class ImagesApi extends BaseApi {
-  public async getImages(query?: any): Promise<any> {
-    return await this.get('/images', undefined, query);
+  public async getImages(query?: any, filterQuery?: string): Promise<any> {
+    return await this.get(`/images${filterQuery ? filterQuery : ''}`, undefined, query);
```
Comment (zlayabekrija) [inline src/api/images.api.ts:5]: Why is get method updated, it should remain as is. Same applies for ```getImages``` method. Please apply suggestions and reach out @MateoKusan98 for help.

File: src/api/images.api.ts [hunk 2]
```diff
@@ -30,4 +30,7 @@ export class ImagesApi extends BaseApi {
   public async myImages(token: string, params?: any): Promise<any> {
     return await this.get('/admin/images', token, params);
   }
+  public async Images(token: string, params?: any): Promise<any> {
```
Comment (zlayabekrija) [inline src/api/images.api.ts:33]: Please follow naming pattern for function names.

File: src/components/ImagePagination/ImagePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        console.log(error);
+      }
+    };
+    getImages();
+  }, [selectedQuery]);
+  const handleLink = (isNext = false) => {
+    if (isNext) {
+      setSelectedQuery(`?${links?.next?.split('?')?.at(-1)}`);
+    } else {
+      setSelectedQuery(`?${links?.last?.split('?')?.at(-1)}`);
```
Comment (zlayabekrija) [inline src/components/ImagePagination/ImagePagination.tsx:39]: Why does going back brings you to the last page
Please refactor this method to receive  a link need without additional logic.
Reach out @MateoKusan98 or @BaneIT for help.

File: src/components/Images/Images.tsx [hunk 4]
```diff
@@ -20,43 +21,46 @@ export default function Images() {
...
+          <input
+            className="form-control"
+            list="datalistOptions"
+            id="exampleDataList"
+            placeholder="Author"
+            name="author"
+            onChange={handleChange}
+            value={formData.author}
+          />
+          <datalist id="datalistOptions">
```
Comment (zlayabekrija) [inline src/components/Images/Images.tsx:48]: Why do we have hardcoded values???

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Please apply requested changes

File: src/components/ImagePagination/ImagePagination.tsx [hunk 5]
```diff
@@ -1,56 +1,42 @@
...
-  current: string;
-  last: string;
-  next: string;
-}
-
-export default function ImagePagination() {
+export default function Home() {
   const {images} = useSelector((state: RootState) => state.images);
   const {user} = useSelector((state: RootState) => state.auth);
+  console.log(images?.links);
```
Comment (zlayabekrija) [inline src/components/ImagePagination/ImagePagination.tsx:12]: Please remove console log

File: src/components/ImagePagination/ImagePagination.tsx [hunk 6]
```diff
@@ -1,56 +1,42 @@
...
-      }
-    };
-    getImages();
-  }, [selectedQuery]);
-  const handleLink = (isNext = false) => {
-    if (isNext) {
-      setSelectedQuery(`?${links?.next?.split('?')?.at(-1)}`);
-    } else {
-      setSelectedQuery(`?${links?.last?.split('?')?.at(-1)}`);
+  const nextDisabled = images?.links?.next === undefined;
```
Comment (zlayabekrija) [inline src/components/ImagePagination/ImagePagination.tsx:14]: Why ???
Passing undefined as an argument should be sufficient

File: src/components/ImagePagination/ImagePagination.tsx [hunk 7]
```diff
@@ -1,56 +1,42 @@
...
+  const prevDisabled = images?.links?.first === undefined;
```
Comment (zlayabekrija) [inline src/components/ImagePagination/ImagePagination.tsx:15]: Same as above

File: src/components/PendingImages/PendingImages.tsx [hunk 8]
```diff
@@ -15,7 +15,7 @@ export default function PendingImages() {
   useEffect(() => {
     const getImages = async () => {
       try {
-        const {data} = await imagesApi.Images(user.token!, {'filter.isApproved': 0});
+        const {data} = await imagesApi.Images(user.token!, {'filter.isApproved': 1});
```
Comment (zlayabekrija) [inline src/components/PendingImages/PendingImages.tsx:18]: Nope, pending images should stay as 0

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Please address give comments, and do not hesitate to reach out for help

@lazarevic-00 @BaneIT @MateoKusan98 please have a look.

File: src/components/Images/Images.tsx [hunk 9]
```diff
@@ -20,43 +21,46 @@ export default function Images() {
...
+            <option />
+            <option />
```
Comment (MateoKusan98) [inline src/components/Images/Images.tsx:50]: Please remove  options if we got no values

**Review by MateoKusan98 — CHANGES_REQUESTED** [review]

**Review by AleksandarGajic01 — APPROVED** [review]
LGTM!

File: src/components/Detail/Detail.tsx [hunk 10]
```diff
@@ -81,31 +84,37 @@ export default function Detail({isAdmin, isDetailsEditPage, idImage}: any) {
...
-                    className="form-check-input"
-                    type="radio"
-                    name="watermark"
-                    checked={currentWatermark === 'InfoBijeljina'}
-                    onChange={handleWatermark}
-                    value="InfoBijeljina"
-                  />
-                  InfoBijeljina
-                </label>
+                {data.map((watermark, key) => {
```
Comment (zlayabekrija) [inline src/components/Detail/Detail.tsx:87]: Nice :+1:

File: src/components/Detail/Detail.tsx [hunk 11]
```diff
@@ -81,31 +84,37 @@ export default function Detail({isAdmin, isDetailsEditPage, idImage}: any) {
...
-                    type="radio"
-                    name="watermark"
-                    id="flexRadioDefault2"
-                    checked={currentWatermark === 'Cafe.ba'}
-                    onChange={handleWatermark}
-                    value="Cafe.ba"
-                  />
-                  Cafe.ba
-                </label>
+                {data.map((watermark, key) => {
```
Comment (zlayabekrija) [inline src/components/Detail/Detail.tsx:103]: Why do we have it twice

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Please merge main branch in yours

**Review by zlayabekrija — APPROVED** [review]

**Pull request merged**